proxy/api/src/main.rs

```diff
@@ -83,15 +83,45 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let bin_dir = config::bin_dir()?;
     coco::git_helper::setup(&proxy_path, &bin_dir).expect("Git remote helper setup failed");
 
-    log::info!("Starting API");
     let ctx = context::Ctx::from(context::Context {
         peer_api,
         keystore,
         store,
     });
+    let watcher_ctx = ctx.clone();
+
+    log::info!("Starting Announcement watcher");
+    tokio::task::spawn_local(async move { announcement_watcher(watcher_ctx).await });
+
+    log::info!("Starting API");
     let api = http::api(ctx, args.test);
 
     warp::serve(api).run(([127, 0, 0, 1], 8080)).await;
 
     Ok(())
 }
+
+async fn announcement_watcher(ctx: context::Ctx) {
+    // TODO(xla): Take interval timings from config/settings.
+    let mut timer = tokio::time::interval(std::time::Duration::from_secs(1));
+
+    loop {
+        timer.tick().await;
+
+        let ctx = ctx.read().await;
+
+        // TODO(xla): get/load old state.
+
+        let old: Vec<announce::Announcement> = vec![];
+        let new = announce::build(&ctx.peer_api).expect("unable to build state");
+        let updates = announce::diff(&old, &new);
```

Comment:
I'm wondering if diff'ing is what we want. The scenario I have in mind is that: if someone was offline during an announcement then they would have to wait for the next change to be made until AND be online again to make sure they have the latest changes.

Reply:
Could you hash out the scenario more going into detail what the availabilities of peer `A` and `B` are?

Wouldn't we expect when a peer comes online again, that they do a full sync with other peers to get to the latest state instead of relying of delta updates of announcements? Or are you advocating to indiscriminately announce all heads every time this routine runs?

Reply:
Gonna resolve this since I'm wrong :)